src/s3path.jl

```diff
@@ -298,7 +305,8 @@ function Base.stat(fp::S3Path)
     s = 0
     last_modified = DateTime(0)
 
-    if exists(fp)
+    # this only works for files and not directories
+    if isfile(fp)
```

Comment:
To clarify, this is because you can only call `s3_get_meta` on objects? So a "directory" object that ends with "/" could also work here?

Reply:
It's pretty cagey... there is really no such thing as a "directory" in S3, but this package pretends like there is to emulate an OS filesystem interface.  If you call `mkdir` with this package, it will create an object for you, but as far as S3 is concerned it's an object like any other.  In those cases, technically this function would work.  However, if you e.g. create `dir/file`, then this will fail for `dir/`.  I'm sure there is more to be done here, but for now, this seems like a workable solution.